readme.md

```diff
@@ -24,110 +33,269 @@ Or you can run it from Maven directly using the Spring Boot Maven plugin. If you
+router:
+```lua
+local function get_vets_with_specialties()
+    local vets_list = {}
+    local vets_by_storage, err =
+    cartridge_pool.map_call('get_vets_with_specialties_id', {}, { uri_list = get_uriList() })
+    if err then
+        return nil, err
+    end
+    for _, vets in pairs(vets_by_storage) do
+        for _, vet in pairs(vets) do
+            local specialties = {}
+            -- one vet may has many specialties
+            for _, specialty_id in pairs(vet[#vet]) do
+                local specialty = crud.get("specialties", specialty_id)
+                table.insert(specialties, crud.unflatten_rows(specialty.rows, specialty.metadata)[1])
+            end
+            -- replace specialty id by specialty name
+            vet[5] = specialties
+            table.insert(vets_list, vet)
+        end
+    end
+    return vets_list
+end
+```
 
-4) Navigate to Petclinic
 
-    Visit [http://localhost:8080](http://localhost:8080) in your browser.
+## Work from java client
 
+Working with tarantool is done using the tarantool function binding via Query annotation, which is indicated above, or use the standard CrudRepository functions (findById, save and etc). 
 
-## Looking for something in particular?
 
-|Spring Boot Configuration | Class or Java property files  |
-|--------------------------|---|
-|The Main Class | [PetClinicApplication](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java) |
-|Properties Files | [application.properties](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/resources) |
-|Caching | [CacheConfiguration](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/system/CacheConfiguration.java) |
+```java
+// Binding tarantool function
+@Query(function = "find_owner_by_id")
+Owner findOwnerById(UUID id);
 
-## Interesting Spring Petclinic branches and forks
+// Using default CrudRepository operations
+Owner save(Owner owner);
+```
 
-The Spring Petclinic "main" branch in the [spring-projects](https://github.com/spring-projects/spring-petclinic)
-GitHub org is the "canonical" implementation, currently based on Spring Boot and Thymeleaf. There are
-[quite a few forks](https://spring-petclinic.github.io/docs/forks.html) in a special GitHub org
-[spring-petclinic](https://github.com/spring-petclinic). If you have a special interest in a different technology stack
-that could be used to implement the Pet Clinic then please join the community there.
+Data from tarantool is mapped from the space description in the form space_object:format to the java description annotated with @Tuple. And the fields themselves are annotated with @Field.
```

Comment:
I suggest to rewrite in more readable way:

Data mapping from tarantool spaces to java entities is implemented with @Tuple and @Field annotations. See tarantool-springdata module for details.